### ISSUE TITLE
Recommended layers for finetuning

### ISSUE THREAD

**Issue by flowtyone** [herimor/voxtream#17]
@herimor awesome work!
I'm attempting to finetune the model to support a few new languages - which layers would you recommend for new language adaptations?

**Comment by herimor** [comment]
Thank you!
You definitely need to re-train [phoneme embeddings](https://github.com/herimor/voxtream/blob/main/voxtream/model.py#L60) layers after adding new phoneme tokens. I would also recommend fine-tuning a [phoneme transformer](https://github.com/herimor/voxtream/blob/main/voxtream/model.py#L50), which is a small model that should not take a lot of computational resources. If the results are not good enough, try to fine-tune the [temporal transformer](https://github.com/herimor/voxtream/blob/main/voxtream/model.py#L53) as well, depending on how much data you are going to use. The depth transformer and audio embeddings/head should be kept frozen, similar to the current training pipeline.

**Comment by flowtyone** [comment]
Thanks - I had some luck with unfreezing the phoneme embeddings, last layer of the phoneme transformer and two last layers of the temporal. I also included the semantic head - do you think this is a good setup? 
The resulted checkpoints definitely speak the required language,  but they're very unstable. I used samples shorter than 55s and repeated padding instead of silence - so I'm thinking that probably had something to do with it.
Single speaker datasets/ datasets with good speaker data are kind of scarce in some languages.
I've been using regular ipa tokens as the tokenizer already uses.

**Comment by herimor** [comment]
I haven't tried multilingual finetuning yet, so it is hard for me to suggest the best layer setup. I suppose the audio length and padding might play a big role. If your audio samples are significantly shorter (5-10 seconds, for example), then training on a similar length, instead of padding to 55s, is a good idea. You can also try to concatenate multiple utterances to achieve a target length of 50-55s. Padding the audio with padding instead of silence should work fine if you ignore these tokens in the loss (which is not implemented by default). If the length padding (by either PAD tokens or silence) exceeds ~5-7% of the average audio length in the training dataset, the shorter training window should be used; this might be the reason behind instabilities.

**Comment by flowtyone** [comment]
Thanks @herimor - I'm trying some datasets with good speaker data so I can properly concatenate and follow the original training recipe, will update if I make any progress 🙏

**Comment by flowtyone** [comment]
@herimor Success! I managed to finetune it to several semantically difficult languages. Unfortunately the data I used is closed, but eventually the recipe above did the trick, along with data preparation that very closely follows the original pretraining recipe - anything else did not work. 

What's amazing about this model is that it is fast enough to run on cpu, and small enough to finetune on a very modest consumer gpu.
